src/comments/CommentItem.js

```diff
@@ -60,6 +60,11 @@ export default class CommentItem extends Component {
     }
   }
 
+  updateCommentPostion = () => {
+    // should happen after react-router updated the route
+    setTimeout(() => this.checkHashLink());
```

Comment:
componentDidMount is not called is just hash is changed. It is for clicking on different comment link for same post eg
From `http://localhost:3000/category/@user/title#hash1` to `http://localhost:3000/category/@user/title#hash2` 

ComponentDidMount is not called in this case. 

Also now checkHashLink is in CommentItem's ComponentDidMount. Shouldn't it would be better to have it in CommentsList. Generally now If user would go to comment's link checkHash would be called as many times as no. of comments and so will `document.getElementById` and `targetElm.scrollIntoView()`